
project device/oneplus/oneplus3/
diff --git a/lineage.mk b/lineage.mk
index 625ca3f..db765cb 100644
--- a/lineage.mk
+++ b/lineage.mk
@@ -27,6 +27,10 @@ $(call inherit-product, vendor/lineage/config/common_full_phone.mk)
 PRODUCT_PROPERTY_OVERRIDES += \
     ro.lineage.build.vendor_security_patch=2018-06-01
 
+# Property for MAC randomization
+PRODUCT_PROPERTY_OVERRIDES += \
+    ro.randomize_mac=1
+
 PRODUCT_NAME := lineage_oneplus3
 PRODUCT_DEVICE := oneplus3
 PRODUCT_MANUFACTURER := OnePlus
