diff --git a/tools/buildinfo.sh b/tools/buildinfo.sh
index 8f5594292..dc2ecd5e8 100755
--- a/tools/buildinfo.sh
+++ b/tools/buildinfo.sh
@@ -16,8 +16,8 @@ echo "ro.build.version.base_os=$PLATFORM_BASE_OS"
 echo "ro.build.date=`$DATE`"
 echo "ro.build.date.utc=`$DATE +%s`"
 echo "ro.build.type=$TARGET_BUILD_TYPE"
-echo "ro.build.user=$USER"
-echo "ro.build.host=`hostname`"
+echo "ro.build.user=$KBUILD_BUILD_USER"
+echo "ro.build.host=$KBUILD_BUILD_HOST"
 echo "ro.build.tags=$BUILD_VERSION_TAGS"
 echo "ro.build.flavor=$TARGET_BUILD_FLAVOR"
 if [ -n "$BOARD_BUILD_SYSTEM_ROOT_IMAGE" ] ; then
